core/vm/evm.go

```diff
@@ -276,6 +297,10 @@ func (evm *EVM) CallCode(caller ContractRef, addr common.Address, input []byte,
 	// It is allowed to call precompiles, even via delegatecall
 	if p, isPrecompile := evm.precompile(addr); isPrecompile {
 		ret, gas, err = RunPrecompiledContract(p, input, gas)
+	} else if evm.isExternalContract(addr) {
+		// TODO: this should execute code of "addr" but use caller as "this" and "caller"
+		//ret, gas, err = evm.Config.ExternalCallback(caller.Address(), addr, value, input, gas)
+		return nil, gas, ErrUnsupportedCall
```

Comment:
For me it is worth it. Maybe it is a good idea to put it like "you can use staticcall and call on native contracts" instead of "you cannot use delegateCall or callcode on native contracts" (half full instead of half empty :) ).

Reply:
no problem if we list explicitly what is not doable, more clear.
can you add it on the other pr?